test/expectations/expectations.go

```diff
@@ -167,6 +167,14 @@ func ExpectDeleted(ctx context.Context, c client.Client, objects ...client.Objec
 	}
 }
 
+func ExpectAlreadyDeletedOrDelete(ctx context.Context, c client.Client, objects ...client.Object) {
+	GinkgoHelper()
+	for _, o := range objects {
+		Expect(c.Delete(ctx, o)).To(Or(Succeed(), MatchError(ContainSubstring("not found"))))
+		Expect(c.Get(ctx, client.ObjectKeyFromObject(o), o)).To(Or(Succeed(), MatchError(ContainSubstring("not found"))))
+	}
```

Comment:
Any reason why we wouldn't want this to be the default behavior of ExpectDeleted()? I also think it'd be nice if we could use the controller runtime client.IgnoreNotFound() here

Reply:
>  why we wouldn't want this to be the default behavior ExpectDeleted()
I dont know if some packages are using the function in the old context where they want the function to imply that object existed and only then delete.